File: dev/lucky_division/lucky_division.cpp
#include <bits/stdc++.h>

auto main() -> int {
  std::string n;
  std::cin >> n;
  std::vector<char> nv(n.begin(), n.end());

  bool flag = true;
  for (auto& i : nv) {
    if (i != '4' && i != '7') {
      flag = false;
    }
  }

  int x = std::stoi(n);
  bool aflag = false;
  std::vector<int> lucky = {4, 7, 44, 47, 74, 77};
  for (auto& l : lucky) {
    if (x % l == 0) {
      aflag = true;
    }
  }

  if (flag || aflag) {
    std::cout << "YES\n";
  } else {
    std::cout << "NO\n";
  }
}

File: dev/lucky_division/lucky_division.py
#!/usr/bin/env python3

def is_almost_lucky(n):
    lucky_numbers = [4, 7, 44, 47, 74, 77]

    for lucky in lucky_numbers:
        if n % lucky == 0:
            return "YES"

    return "NO"

n = list(input())
flag = True
for i in n:
    if i not in '47':
        flag = False
x = int(''.join(n))

print("YES" if flag == True else is_almost_lucky(x))
